docs/adrs/00009-new-field-license.md

```diff
@@ -0,0 +1,160 @@
+
+- ID: This is a standard SPDX license ID.
+- License expression: Similar to SPDX, it supports license expressions.
+- Name: This is similar to a custom license reference, allowing for user-defined license names.
+
+In Trustify, the handling is as follows:
+
+When processing SPDX, Trustify parses the license expression into standard SPDX license IDs and exceptions, and stores them in spdx_licenses and spdx_license_exceptions. All custom license references (LicenseRef) are filtered out, while the original expression is preserved in the text field.
+
+When processing CycloneDX, there are three scenarios:
+
+- If the license is an ID, the value is saved directly in the text field and spdx_licenses.
+- If the license is a name: The value is saved directly in the text field, and both spdx_licenses and spdx_license_exceptions remain empty.
+- If the license is an expression: Trustify parses it into standard SPDX license IDs and exceptions, stores them in spdx_licenses and spdx_license_exceptions, and also preserves the original expression in the text field.
+
+## The Current Issue
+
+- In CycloneDX, when the license type is 'name', Trustify only stores it in a text field, and we cannot determine whether this field represents just a name.
+- In SPDX, when a license expression contains a custom license ID, Trustify also saves it only in a text field, and filters out the custom license ID during the process of parsing it into standard SPDX license IDs and exceptions.
+- In SPDX, when a license expression includes a custom license ID, the custom license ID itself is meaningless and only has significance within the current SBOM, so it cannot be used for statistics or searches.
+
+## Decision
+
+Add a new field named `custom_license_refs` to the license table. Its type is a list, which includes licenses of type name appearing in CycloneDX, as well as license expressions that contain custom licenses in SPDX.
+
+```sql
+ALTER TABLE license ADD COLUMN custom_license_refs text[];
+```
+
+### CycloneDX
+
+Licenses of type 'name':
+
+```json
+{
+  "bom-ref": "pkg:rpm/centos/audit-libs@3.0-0.17.20191104git1c2f876.el8?arch=x86_64&upstream=audit-3.0-0.17.20191104git1c2f876.el8.src.rpm&distro=centos-8&package-id=05587266ec4157c9",
+  "type": "library",
+  "publisher": "CentOS",
+  "name": "audit-libs",
+  "version": "3.0-0.17.20191104git1c2f876.el8",
+  "licenses": [
+    {
+      "license": {
+        "name": "LGPLv2+"
+      }
+    }
+  ],
+  "cpe": "cpe:2.3:a:audit-libs:audit-libs:3.0-0.17.20191104git1c2f876.el8:*:*:*:*:*:*:*",
+  "purl": "pkg:rpm/centos/audit-libs@3.0-0.17.20191104git1c2f876.el8?arch=x86_64&upstream=audit-3.0-0.17.20191104git1c2f876.el8.src.rpm&distro=centos-8",
+  "properties": [
+    {
+      "name": "syft:package:foundBy",
+      "value": "rpm-db-cataloger"
+    }
+  ]
+}
+```
+
+`custom_license_refs` should be a JSONB mapping all custom license identifiers to their descriptions or SPDX expressions.
+
+Example:
+
+```json
+{
```

Comment:
What would be the other fields of that table entry?

Reply:
Maybe you can add those to the example

Reply:
updated, Please check again.